tobac/merge_split.py

```diff
@@ -0,0 +1,477 @@
+                )
+                if d <= distance:
+                    a_points.append([a_pointx, a_pointy])
+                    b_points.append([b_pointx, b_pointy])
+                    dist.append(d)
+                    a_names.append(i)
+                    b_names.append(j)
+
+    id = []
+    for i in range(len(dist) - 1, -1, -1):
+        if a_names[i] == b_names[i]:
+            id.append(i)
+            a_points.pop(i)
+            b_points.pop(i)
+            dist.pop(i)
+            a_names.pop(i)
+            b_names.pop(i)
+        else:
+            continue
+
+    g = Graph()
+    for i in np.arange(len(dist)):
+        g.add_edge(a_names[i], b_names[i], weight=dist[i])
+
+    tree = minimum_spanning_edges(g)
+    tree_list = list(minimum_spanning_edges(g))
+
+    new_tree = []
+    for i, j in enumerate(tree_list):
+        frame_a = np.nanmax(track_groups[j[0]].frame.values)
+        frame_b = np.nanmin(track_groups[j[1]].frame.values)
+        if np.abs(frame_a - frame_b) <= frame_len:
+            new_tree.append(tree_list[i][0:2])
+    new_tree_arr = np.array(new_tree)
+
+    TRACK["cell_parent_track_id"] = np.zeros(len(TRACK["cell"].values))
+    cell_id = np.unique(
+        TRACK.cell.values.astype(float)[~np.isnan(TRACK.cell.values.astype(float))]
+    )
+    track_id = dict()  # same size as number of total merged tracks
+
+    arr = np.array([0])
+    for p in cell_id:
+        j = np.where(arr == int(p))
+        if len(j[0]) > 0:
+            continue
+        else:
+            k = np.where(new_tree_arr == p)
+            if len(k[0]) == 0:
+                track_id[p] = [p]
+                arr = np.append(arr, p)
+            else:
+                temp1 = list(np.unique(new_tree_arr[k[0]]))
+                temp = list(np.unique(new_tree_arr[k[0]]))
+
+                for l in range(15):
+                    for i in temp1:
+                        k2 = np.where(new_tree_arr == i)
+                        temp.append(list(np.unique(new_tree_arr[k2[0]]).squeeze()))
+                        temp = list(flatten(temp))
+                        temp = list(np.unique(temp))
+
+                    if len(temp1) == len(temp):
+                        break
+                    temp1 = np.array(temp)
+
+                for i in temp1:
+                    k2 = np.where(new_tree_arr == i)
+                    temp.append(list(np.unique(new_tree_arr[k2[0]]).squeeze()))
+
+                temp = list(flatten(temp))
+                temp = list(np.unique(temp))
+                arr = np.append(arr, np.unique(temp))
+
+                track_id[np.nanmax(np.unique(temp))] = list(np.unique(temp))
+
+    storm_id = [0]  # default because we don't track larger storm systems *yet*
+    print("found storm id")
+
+    track_parent_storm_id = np.repeat(
+        0, len(track_id)
+    )  # This will always be zero when we don't track larger storm systems *yet*
+    print("found track parent storm ids")
```

Comment:
I think generally we should avoid debug statements in code, but if we have them, we should use `logging.debug` as in the other modules.